Parser/Node.cs

```diff
@@ -48,6 +53,36 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
 public record NAssignStmt (Token Name, NExpr Expr) : NStmt {
    public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
 }
+
+// A Read statement
+public record NReadStmt (Token[] Args) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// A Call statement
+public record NCallStmt (Token Name, NExpr[] Args) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+// If statement
+public record NIfStmt (NExpr Expr, NStmt[] Stmts) : NStmt {

```

Comment:
This should just be:
`public record NIfStmt (NExpr Expr, NStmt IfPart, NStmt? ElsePart)`
There is one (mandatory) statement for the if part, and an optional statement for the else part. If there are multiple statments in the If, that is basically a single CompoundStmt bracketed by begin..end